spec/queueSpec.js

```diff
@@ -0,0 +1,69 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import Queue from '../src/queue'
+
+chai.use(chaiChange)
+describe('Queue', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(Queue).to.be.a('function')
+  })
+
+  context('enqueue(element)', () => {
+    it('pushes an element to the back of the list.', () => {
+      const myQueue = new Queue()
+
+      expect(() => myQueue.enqueue('foo')).to.alter(() => myQueue.length(), { from: 0, to: 1 })
+
+      expect(() => myQueue.enqueue('boo')).to.alter(() => myQueue.length(), {from: 1, to: 2})
+
+      expect(() => myQueue.enqueue('moo')).to.alter(() => myQueue.length(), {from: 2, to: 3})
+    })
+  })
+
+  context('dequeue()',() => {
+    it('removes elements from the front of list.', () => {
+      const myQueue = new Queue()
+      myQueue.enqueue('foo')
+      myQueue.enqueue('boo')
+      expect(() => myQueue.dequeue().to.equal('boo'));
+    })
+  })
+
+  context('length()', () => {
+    it('returns the length of the array', () => {
+      const myQueue = new Queue()
+      myQueue.enqueue('foo')
+      expect(() => myQueue.length(), { from: 0, to: 1 })
+    })
+  })
+
+  context('front()',() => {
+    it('returns the first element in the list and returns null if the value is null', () => {
+      const myQueue = new Queue()
```

Comment:
test descriptions that have the words "and" or "or" in them are an indication that you should split this up into two tests to isolate cases
`context('front()', () => {
    context('when the queue is not empty', () => {
      it('returns the front element', () => {

        myQueue.enqueue('foo')
        myQueue.enqueue('bar')
        expect(myQueue.front()).to.be.an.instanceof(Node)
      })
    })

    context('when the queue is empty', () => {
      it('returns null', () => {

        expect(myQueue.front()).to.equal(null)
      })
    })
  })
`